packages/react-dom/src/events/plugins/ModernSelectEventPlugin.js

```diff
@@ -168,13 +169,19 @@ const SelectEventPlugin = {
     eventSystemFlags,
     container,
   ) {
-    const containerOrDoc =
-      container || getEventTargetDocument(nativeEventTarget);
+    const doc = getEventTargetDocument(nativeEventTarget);
     // Track whether all listeners exists for this plugin. If none exist, we do
     // not extract events. See #3639.
     if (
-      !containerOrDoc ||
-      !isListeningToAllDependencies('onSelect', containerOrDoc)
+      // We only listen to TOP_SELECTION_CHANGE on the document, never the
+      // root.
+      !isListeningToEvent(TOP_SELECTION_CHANGE, doc) ||
+      // If we are handling TOP_SELECTION_CHANGE, then we don't need to
+      // check for the other dependencies, as TOP_SELECTION_CHANGE is only
+      // event attached from the onChange plugin and we don't expose an
+      // onSelectionChange event from React.
+      (topLevelType !== TOP_SELECTION_CHANGE &&
+        !isListeningToEvents(rootTargetDependencies, container))
     ) {
       return null;
```

Comment:
It feels weird that we have to do this test here and not in other event plugins. 🤷‍♂️

Reply:
We don't do the `isListeningToEvents` logic check in other plugins.